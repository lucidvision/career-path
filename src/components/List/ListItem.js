import React from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

ListItem.propTypes = {
  item: PropTypes.object.isRequired
}

export default function ListItem ({item, onItemPressed}) {
  return (
    <TouchableOpacity onPress={() => onItemPressed(item)}>
      <View style={styles.item}>
        <Text>{item.name}</Text>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  item: {
    backgroundColor: '#ffffff',
    height: 100,
    justifyContent: 'center',
    padding: 10
  }
})
