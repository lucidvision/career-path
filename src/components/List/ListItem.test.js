import React from 'react'
import { shallow } from 'enzyme'
import ListItem from './ListItem'

const createListItem = (override) => {
  const props = {
    item: {name: 'Rainbow'},
    onItemPressed: () => {},
    ...override
  }

  return shallow(<ListItem {...props} />)
}

describe('ListItem', () => {
  const component = createListItem()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('triggers callback on press', () => {
    const mockOnItemPressed = jest.fn()
    const component = createListItem({onItemPressed: mockOnItemPressed})
    const touchableOpacity = component.find('TouchableOpacity')
    touchableOpacity.props().onPress()
    expect(mockOnItemPressed).toHaveBeenCalledWith({name: 'Rainbow'})
  })
})
