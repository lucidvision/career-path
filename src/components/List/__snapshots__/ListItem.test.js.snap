// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "justifyContent": "center",
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Rainbow
    </Text>
  </View>
</TouchableOpacity>
`;
